pkg/sql/schemachanger/scbuild/internal/scbuildstmt/helpers.go

```diff
@@ -292,6 +292,41 @@ func getSortedColumnIDsInIndex(
 	return keyColumnIDs, keySuffixColumnIDs, storingColumnIDs
 }
 
+// getNonDropColumnIDs returns all non-drop column IDs in a table in ascending order.
+func getNonDropColumnIDs(b BuildCtx, tableID catid.DescID) (ret []catid.ColumnID) {
+	scpb.ForEachColumn(b.QueryByID(tableID).Filter(publicTargetFilter), func(
+		_ scpb.Status, _ scpb.TargetStatus, e *scpb.Column,
+	) {
+		ret = append(ret, e.ColumnID)
+	})
+	sort.Slice(ret, func(i, j int) bool {
+		return ret[i] < ret[j]
+	})
+	return ret
+}
+
+// getColumnIDFromColumnName looks up a column's ID by its name.
+// If no column with this name exists, 0 will be returned.
+func getColumnIDFromColumnName(
+	b BuilderState, tableID catid.DescID, columnName tree.Name,
+) catid.ColumnID {
+	colElems := b.ResolveColumn(tableID, columnName, ResolveParams{
+		IsExistenceOptional: true,
+		RequiredPrivilege:   privilege.CREATE,
+	})
```

Comment:
Using `ResolveColumn ` will not work on columns added in the same transaction:
```
BEGIN;
ALTER TABLE t ADD COLUMN c INT;
ALTER TABLE t ADD CONSTRAINT ck CHECK (c > 0);
```
While this case is not supported in prod, we should try to support it in all `pkg/sql/schemachanger/...` tests.

Reply:
Oh nice. I thought otherwise for some reason. Excellent! That's a relief.